.github/workflows/comment-diffs.yml

```diff
@@ -0,0 +1,306 @@
+
+              # Get new version remote commit hash
+              # new_version_commit_hash=$(git rev-parse --short origin/diffs)
+
+              # Get old version remote commit hash
+              # old_version_commit_hash=$(git rev-parse --short origin/diffs~)
+
+              # Add output to file
+              # echo "[$library_type/$language](https://github.com/${{github.repository}}/compare/$old_version_commit_hash..$new_version_commit_hash)" >> "$output_file"
+ 
+              # Write the diff to the output file
+              git diff HEAD~ >> $output_file
+            fi
+          }
+
+          for library_type in "${libraryTypes[@]}"; do
+            for language in "${languages[@]}"; do
+              if [[ ! "${exclude[*]}" =~ ${library_type}/${language} ]]; then
+                copy_commit_diff "$library_type" "$language"
+              fi
+            done
+          done
+
+          if ls $output_path/*.txt; then
+            echo "DIFF_IS_NON_EMPTY=1" >> $GITHUB_OUTPUT
+          else
+            echo "DIFF_IS_NON_EMPTY=0" >> $GITHUB_OUTPUT
+          fi
+
+          # Remove outputs folder in cwd if it exists
+          rm -rf ./outputs || true
+
+          # Copy the outputs to cwd
+          cp -r $output_path ./outputs
+
+      - uses: actions/upload-artifact@v3
+        if: ${{ steps.create-diffs.outputs.DIFF_IS_NON_EMPTY == '1' }}
+        with:
+          name: outputs
+          path: outputs/*.txt
+
+  read-artifacts-and-comment:
+    name: Read the artifacts and comment
+    runs-on: ubuntu-latest
+    needs: [generate-diffs-if-needed]
+    steps:
+      - name: Load outputs
+        uses: actions/download-artifact@v3
+        continue-on-error: true
+        with:
+          name: outputs
+          path: outputs
+
+      - name: Output artifacts
+        id: artifacts
+        run: |
+          if ls outputs/*.txt; then
+            result=1 # Artifacts are not empty
+          else
+            result=0 # Artifacts are empty
+          fi
+
+          echo "ARTIFACTS_ARE_NON_EMPTY=$result" >> $GITHUB_OUTPUT
+
+      - name: Comment on PR
+        if: ${{ steps.artifacts.outputs.ARTIFACTS_ARE_NON_EMPTY == '1' }}
+        uses: actions/github-script@v3
+        with:
+          github-token: ${{secrets.GITHUB_TOKEN}}
+          script: |
+            const fs = require('fs');
+
+            const replaceTable = {
+              "module-legacy": "Native module",
+              "module-mixed": "Turbo module with backward compat",
+              "module-new": "Turbo module",
+              "view-legacy": "Native view",
+              "view-mixed": "Fabric view with backward compat",
+              "view-new": "Fabric view",
+              "java-objc": "Java and Objective C",
+              "java-swift": "Java and Swift",
+              "kotlin-objc": "Kotlin and Objective C",
+              "kotlin-swift": "Kotlin and Swift",
+              ".txt": ""
+            }
```

Comment:
having the matrix in JSON would be nice, since this table could also be colocated.

though maybe we should add a command to `create-react-native-library`, `create-react-native-library --list-types` and `create-react-native-library --list-languages` or something that lists all available templates instead hardcoding. then we could use that for matrix in build-templates as well as here. i have forgotten to change the matrix when i change templates in past (worse when something gets added and there's no test failure), and it'll avoid this.